TodoListManagerTest/src/test/java/PerformanceTest/TodosPerformanceTest.java

```diff
@@ -0,0 +1,177 @@
+        long[] freeMemory = new long[populationSize.length];
+        int targetIndex = 0;
+
+        for(int i=1; i<=max;i++){
+            JSONObject body = new JSONObject();
+            String newTitle = "New Todo " + i;
+            String newDescription = "New Description";
+            boolean newDoneStatus = false;
+            body.put("title",newTitle);
+            body.put("doneStatus",newDoneStatus);
+            body.put("description",newDescription);
+            long startTime = System.nanoTime();
+            Response response = RestAssured.given().contentType(ContentType.JSON).body(body.toString()).post(pUrl);
+            long endTime = System.nanoTime();
+            if(i == populationSize[targetIndex]){
+                double cpuLoad = osBean.getCpuLoad() * 100;
+                long memory = osBean.getFreeMemorySize();
+                time[targetIndex] = (endTime-startTime);
+                cpuUsage[targetIndex] = cpuLoad;
+                freeMemory[targetIndex] = memory;
+                targetIndex++;
+            }
+        }
+        System.out.println("---Add TODOS---");
+        for (int k = 0; k < populationSize.length; k++) {
+            System.out.println("Time for size " + populationSize[k] + ": " + time[k] + "ns");
+            System.out.println("CPU usage for size " + populationSize[k] + ": " + cpuUsage[k] + "%");
+            System.out.println("Available free memory for size " + populationSize[k] + ": " + freeMemory[k] + "bytes");
+        }
+    }
+
+    @Test
+    public void UpdateTodosJSONPut(){
+        int max = populationSize[populationSize.length - 1];
+        long[] time = new long[populationSize.length];
+        double[] cpuUsage = new double[populationSize.length];
+        long[] freeMemory = new long[populationSize.length];
+        int targetIndex = 0;
+
+        for(int i=1; i<=max;i++){
+            JSONObject body = new JSONObject();
+            String newTitle = "New Todo " + i;
+            String newDescription = "New Description";
+            boolean newDoneStatus = false;
+            body.put("title",newTitle);
+            body.put("doneStatus",newDoneStatus);
+            body.put("description",newDescription);
+            Response response = RestAssured.given().contentType(ContentType.JSON).body(body.toString()).post(pUrl);
+        }
+
+        for(int i=1; i<=max;i++){
+            JSONObject body = new JSONObject();
+            String newTitle = "New Todo " + i;
+            String newDescription = "New Description";
+            boolean newDoneStatus = true;
+            body.put("title",newTitle);
+            body.put("doneStatus",newDoneStatus);
+            body.put("description",newDescription);
+            long startTime = System.nanoTime();
+            Response response = RestAssured.given().contentType(ContentType.JSON).body(body.toString()).put(pUrl+"/"+i);
+            long endTime = System.nanoTime();
```

Comment:
Here you got the logic wrong, in this case you are populating 1000 objects and then you are updating every object and measuring the time of them. What you need to do is to populate and while populating, you update the todo at each target.

Reply:
Done